.github/workflows/format.yml

```diff
@@ -0,0 +1,51 @@
+name: Format-manifest checks
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/**
+      - manifests/**
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/**
+      - manifests/**
+
+jobs:
+  format-manifest:
+    name: format-manifest test
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: packages-index
+
+      - name: Install ruyi
+        run: |
+          mkdir bin/
+
+          version="$(curl --silent https://api.github.com/repos/ruyisdk/ruyi/releases/latest | \
+            jq --raw-output ".tag_name")"
+          curl -sL -o bin/ruyi "https://github.com/ruyisdk/ruyi/releases/download/${version}/ruyi-${version}.amd64"
+
+          chmod +x ./bin/ruyi
+          ./bin/ruyi version
+          ./bin/ruyi telemetry consent
+          ./bin/ruyi telemetry upload
```

Comment:
Could be better to perform the upload in a post-action step.

Reply:
I move the upload action to a separate step and it will always upload telemetry data after test.